registry/storage/catalog.go

```diff
@@ -71,33 +62,69 @@ func (reg *registry) Repositories(ctx context.Context, repos []string, last stri
 
 // Enumerate applies ingester to each repository
 func (reg *registry) Enumerate(ctx context.Context, ingester func(string) error) error {
-	repoNameBuffer := make([]string, 100)
+	repos := make([]string, 100)
```

Comment:
We don't really need to maintain this buffer since we aren't calling `Repositories`. The `Walk` function can call the `ingester` directly.